nifi-nar-bundles/nifi-snmp-bundle/nifi-snmp-processors/src/test/java/org/apache/nifi/snmp/operations/SNMPGetterTest.java

```diff
@@ -0,0 +1,174 @@
+        SNMPGetter getter = new SNMPGetter(snmp, target, new OID("1.3.6.1.4.1.32437"));
+        final List<TreeEvent> responseEvents = getter.walk();
+        assertEquals(OIDValue1, responseEvents.get(0).getVariableBindings()[0].getVariable().toString());
+        assertEquals(OIDValue2, responseEvents.get(1).getVariableBindings()[0].getVariable().toString());
+
+    }
+
+    @Test
+    public void testSuccessfulSnmpV2Get() throws IOException {
+        Snmp snmp = SNMPTestUtils.createSnmp();
+        CommunityTarget target = SNMPTestUtils.createCommTarget("public", LOCALHOST + "/" + snmpV2Agent.getPort(), SnmpConstants.version2c);
+        SNMPGetter getter = new SNMPGetter(snmp, target, readOnlyOID1);
+        ResponseEvent response = getter.get();
+        assertEquals(OIDValue1, response.getResponse().get(0).getVariable().toString());
+
+    }
+
+    @Test
+    public void testSuccessfulSnmpV2Walk() throws IOException {
+        Snmp snmp = SNMPTestUtils.createSnmp();
+        CommunityTarget target = SNMPTestUtils.createCommTarget("public", LOCALHOST + "/" + snmpV2Agent.getPort(), SnmpConstants.version2c);
+        SNMPGetter getter = new SNMPGetter(snmp, target, new OID("1.3.6.1.4.1.32437"));
+        final List<TreeEvent> responseEvents = getter.walk();
+        final VariableBinding[] variableBindings = responseEvents.get(0).getVariableBindings();
+        assertEquals(OIDValue1, variableBindings[0].getVariable().toString());
+        assertEquals(OIDValue2, variableBindings[1].getVariable().toString());
+
+    }
+
+    @Test
+    public void testSuccessfulSnmpV3Get() throws IOException {
+        SNMP4JSettings.setForwardRuntimeExceptions(true);
+        Snmp snmp = SNMPTestUtils.createSnmp();
+        final UserTarget userTarget = SNMPTestUtils.prepareUser(snmp, LOCALHOST + "/" + snmpV3Agent.getPort(), SecurityLevel.AUTH_NOPRIV,
+                "SHA", AuthSHA.ID, null, "SHAAuthPassword", null);
+        SNMPGetter getter = new SNMPGetter(snmp, userTarget, readOnlyOID1);
+        ResponseEvent response = getter.get();
+        assertEquals(OIDValue1, response.getResponse().get(0).getVariable().toString());
+
+    }
+
+    @Test
+    public void testUnauthorizedUserSnmpV3GetReturnsNull() throws IOException {
+        SNMP4JSettings.setForwardRuntimeExceptions(true);
+        Snmp snmp = SNMPTestUtils.createSnmp();
+        final UserTarget userTarget = SNMPTestUtils.prepareUser(snmp, LOCALHOST + "/" + snmpV3Agent.getPort(), SecurityLevel.AUTH_NOPRIV,
+                "FakeUserName", AuthSHA.ID, null, "FakeAuthPassword", null);
+        SNMPGetter getter = new SNMPGetter(snmp, userTarget, readOnlyOID1);
+        ResponseEvent response = getter.get();
+        assertEquals("Null", response.getResponse().get(0).getVariable().toString());
+
+    }
+
+    @Test
+    public void testSnmpV1GetTimeoutReturnsNull() throws IOException {
+        Snmp snmp = SNMPTestUtils.createSnmp();
```

Comment:
Minor: there are a lot of duplicated part. Would it make sense to extract them? I think it would make the actual tests more compact and readable.